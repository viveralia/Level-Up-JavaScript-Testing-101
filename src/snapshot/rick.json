{
  "id": 1,
  "name": "Rick Sanchez",
  "status": "alive",
  "species": "Human",
  "gender": "male"
}
