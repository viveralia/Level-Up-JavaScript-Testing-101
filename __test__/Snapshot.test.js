import { getCharacter } from '../src/snapshot/Snapshot'
import rick from '../src/snapshot/rick.json'

describe('Es hora de las instantáneas', () => {
  // La primera vez que se ejecuta este código, se guardará el resultado de la petición como un snapshot. Posteriormente, se comparará el snapshot con nuevas peticiones para asegurarse de que la respuesta no ha cambiado. Para actualizar el snapshot es necesario correr el comando: jest -u
  test('Snapshot', () => {
    expect(getCharacter(rick)).toMatchSnapshot()
  })

  // Estos valores siempre estarán cambiando porque son aleatorios, al igual que sería un updatedAt
  test('Siempre fallará la instantánea', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20)
    }
    expect(user).toMatchSnapshot()
  })

  test('Tenemos una exepción', () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: 'Iván Viveros'
    }

    // Dentro de los parámetros se puede especificar que algún valor va a cambiar especificando el tipo de valor que esperamos recibir (no funcionó, lol)
    expect(user).toMatchSnapshot({
      id: expect.any(Number)
    })
  })
})
