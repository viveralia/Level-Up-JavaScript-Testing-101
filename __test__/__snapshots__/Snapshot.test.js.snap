// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantáneas Siempre fallará la instantánea 1`] = `
Object {
  "createdAt": 2019-12-02T21:43:18.194Z,
  "id": 12,
}
`;

exports[`Es hora de las instantáneas Snapshot 1`] = `
Object {
  "gender": "male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "alive",
}
`;

exports[`Es hora de las instantáneas Tenemos una exepción: [object Object] 1`] = `
Object {
  "id": 19,
  "name": "Iván Viveros",
}
`;
